rules-tests/Php80/Rector/Switch_/ChangeSwitchToMatchRector/Fixture/no_default_return_different.php.inc

```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace Rector\Tests\Php80\Rector\Switch_\ChangeSwitchToMatchRector\Fixture;
+
+class NoDefaultReturnDifferent
+{
+    public function run($value)
+    {
+        if (strlen($value) < 3) {
+            switch($value) {
+                case 'A':
+                case 'B':
+                    return 'AB';
+                case 'C':
+                    return 'D';
+
+            }
+        }
+
+        return 'test';
+    }
+}
+
+?>
+-----
+<?php
+
+namespace Rector\Tests\Php80\Rector\Switch_\ChangeSwitchToMatchRector\Fixture;
+
+class NoDefaultReturnDifferent
+{
+    public function run($value)
+    {
+        if (strlen($value) < 3) {
+            return match ($value) {
+                'A', 'B' => 'AB',
+                'C' => 'D',
+                default => 'test',
+            };
+        }
+
+        return 'test';
+    }
+}
+
+?>
```

Comment:
This one should be still skipped. The next expression should not be taken info account, only the scope of `switch` itself.

The `$value` can be dynamic and the results will be different. 

```php
        if (strlen($value) < 3) {
            return match ($value) {
                'A', 'B' => 'AB',
                'C' => 'D',
                default => $value,
            };
        }

        return $value;
```